nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java

```diff
@@ -0,0 +1,162 @@
+import org.projectnessie.versioned.persist.tests.extension.NessieDbAdapter;
+import org.projectnessie.versioned.persist.tests.extension.NessieDbAdapterName;
+import org.projectnessie.versioned.persist.tests.extension.NessieExternalDatabase;
+
+@ExtendWith(DatabaseAdapterExtension.class)
+@NessieDbAdapterName(InmemoryDatabaseAdapterFactory.NAME)
+@NessieExternalDatabase(InmemoryTestConnectionProviderSource.class)
+public class TestNessieCatalog extends CatalogTests<NessieCatalog> {
+
+  @NessieDbAdapter(storeWorker = TableCommitMetaStoreWorker.class)
+  static DatabaseAdapter databaseAdapter;
+
+  @RegisterExtension
+  static NessieJaxRsExtension server = new NessieJaxRsExtension(() -> databaseAdapter);
+
+  @TempDir public Path temp;
+
+  private NessieCatalog catalog;
+  private NessieApiV1 api;
+  private Configuration hadoopConfig;
+  private String uri;
+
+  @BeforeEach
+  public void beforeEach(@NessieUri URI nessieUri) throws IOException {
+    this.uri = nessieUri.toString();
+    this.api = HttpClientBuilder.builder().withUri(this.uri).build(NessieApiV1.class);
+
+    resetData();
+
+    try {
+      api.createReference().reference(Branch.of("main", null)).create();
+    } catch (Exception e) {
+      // ignore, already created. Can't run this in BeforeAll as quarkus hasn't disabled auth
+    }
+
+    hadoopConfig = new Configuration();
+    catalog = initNessieCatalog("main");
+  }
+
+  @AfterEach
+  public void afterEach() throws NessieConflictException, NessieNotFoundException {
+    resetData();
+    try {
+      if (catalog != null) {
+        catalog.close();
+      }
+      api.close();
+    } finally {
+      catalog = null;
+      api = null;
+      hadoopConfig = null;
+    }
+  }
+
+  private void resetData() throws NessieConflictException, NessieNotFoundException {
+    for (Reference r : api.getAllReferences().get().getReferences()) {
+      if (r instanceof Branch) {
+        api.deleteBranch().branch((Branch) r).delete();
+      } else {
+        api.deleteTag().tag((Tag) r).delete();
+      }
+    }
+    api.createReference().reference(Branch.of("main", null)).create();
+  }
+
+  private NessieCatalog initNessieCatalog(String ref) {
+    NessieCatalog newCatalog = new NessieCatalog();
+    newCatalog.setConf(hadoopConfig);
+    newCatalog.initialize(
+        "nessie",
+        ImmutableMap.of(
+            "ref",
+            ref,
+            CatalogProperties.URI,
+            uri,
+            "auth-type",
+            "NONE",
+            CatalogProperties.WAREHOUSE_LOCATION,
+            temp.toUri().toString()));
+    return newCatalog;
+  }
+
+  @Override
+  protected NessieCatalog catalog() {
+    return catalog;
+  }
+
+  @Override
+  protected boolean supportsNamespaceProperties() {
+    return false;
+  }
+
+  @Override
+  protected boolean supportsServerSideRetry() {
+    // TODO: we do support retries, but a bunch of tests are currently failing
```

Comment:
I think you'd have to update to the diff-based commits to make this work. If you end up using the REST catalog API that would be a nice feature to pull in at the same time.